skfp/model_selection/utils.py

```diff
@@ -63,29 +95,61 @@ def split_additional_data(
     )
 
 
-def validate_train_valid_test_split_sizes(
-    train_size: Optional[float],
-    valid_size: Optional[float],
-    test_size: Optional[float],
+def validate_and_scale_train_valid_test_split_sizes(
+    train_size: Optional[Union[float, int]],
+    valid_size: Optional[Union[float, int]],
+    test_size: Optional[Union[float, int]],
+    data_length: int,
 ) -> tuple[float, float, float]:
     """
     Ensure the sum of train_size, valid_size, and test_size equals 1.0 and provide default values if necessary.
+    If test_size, valid_size and train_size are floats, this method returns scaled data_length.
+    If test_size, valid_size and train_size are ints, this method returns the provided sizes.
     """
     if train_size is None and valid_size is None and test_size is None:
-        return 0.8, 0.1, 0.1
+        train_size, valid_size, test_size = 0.8, 0.1, 0.1
 
     if train_size is None or valid_size is None or test_size is None:
-        raise ValueError(
-            "All of train_size, valid_size, and test_size must be provided."
-        )
+        raise ValueError("All of the sizes must be provided.")
+
+    sizes: tuple[Union[float, int], Union[float, int], Union[float, int]] = (
+        train_size,
+        valid_size,
+        test_size,
+    )
 
-    if valid_size == 0.0:
-        warnings.warn(
-            "Validation set will not be returned since valid_size was set to 0.0."
-            "Consider using train_test_split instead."
+    if not all(isinstance(size, (int, float)) for size in sizes):
+        raise ValueError("All sizes must be either int or float.")
+
+    if not all(isinstance(size, type(sizes[0])) for size in sizes):
+        raise ValueError("All sizes must be of the same type.")
+
+    if isinstance(sizes[0], float):
+        if any(size < 0 for size in sizes):
+            raise ValueError("Sizes as floats must be non-negative.")
```

Comment:
We assume it is so here, since this function is already called after a function with validator. You already make this assumption in train-test split